src/products/components/ProductListPage/ProductListPage.tsx

```diff
@@ -185,50 +194,95 @@ export const ProductListPage: React.FC<ProductListPageProps> = props => {
-          allTabLabel={intl.formatMessage({
-            id: "aFLtLk",
-            defaultMessage: "All Products",
-            description: "tab name",
-          })}
-          filterStructure={filterStructure}
-          searchPlaceholder={intl.formatMessage({
-            id: "kIvvax",
-            defaultMessage: "Search Products...",
-          })}
-        />
-        <ProductListDatagrid
-          {...listProps}
-          filterDependency={filterDependency}
-          activeAttributeSortId={activeAttributeSortId}
-          columnQuery={columnQuery}
-          defaultSettings={defaultSettings}
-          availableInGridAttributes={availableInGridAttributes}
-          loading={loading}
-          hasMore={hasMore}
-          gridAttributes={gridAttributes}
-          onColumnQueryChange={onColumnQueryChange}
-          onFetchMore={onFetchMore}
-          products={listProps.products}
-          settings={settings}
-          selectedChannelId={selectedChannelId}
-          onUpdateListSettings={onUpdateListSettings}
-          onRowClick={id => {
-            navigate(productUrl(id));
-          }}
-        />
+        <Box
+          display="flex"
+          flexDirection="column"
+          width="100%"
+          alignItems="stretch"
+          justifyContent="space-between"
+        >
+          <FilterBar
+            withoutBorder
+            currencySymbol={currencySymbol}
+            currentTab={currentTab}
+            initialSearch={initialSearch}
+            onAll={onAll}
+            onFilterChange={onFilterChange}
+            onFilterAttributeFocus={onFilterAttributeFocus}
+            onSearchChange={onSearchChange}
+            onTabChange={onTabChange}
+            onTabDelete={onTabDelete}
+            onTabSave={onTabSave}
+            tabs={tabs}
+            allTabLabel={intl.formatMessage({
+              id: "aFLtLk",
+              defaultMessage: "All Products",
+              description: "tab name",
+            })}
+            filterStructure={filterStructure}
+            searchPlaceholder={intl.formatMessage({
+              id: "kIvvax",
+              defaultMessage: "Search Products...",
+            })}
+          />
+          {/* Temporary solution until header is reworked */}
+          <Button
+            variant="tertiary"
+            onClick={() => setDatagridView(state => !state)}
+          >
+            {!isDatagridView ? (
+              <>
+                <TableEditIcon />
+                <FormattedMessage
+                  id="WACLJx"
+                  defaultMessage="Switch to datagrid view"
+                  description="switch"
+                />
+              </>
+            ) : (
+              <>
+                <ProductsIcons />
+                <FormattedMessage
+                  id="CU+aFW"
+                  defaultMessage="Switch to tile view"
+                  description="switch"
+                />
+              </>
+            )}
+          </Button>
```

Comment:
question: WDYT about move this to seprate component?

Reply:
This is going to be removed in the next PR - we don't want a button that switches views, we want an actual switch

Reply:
+1 In the next PRs please try to break it down into smaller components